### DISCUSSION TITLE
Erklärung zu Port 8400 und Unterschiede zu 8082

### DISCUSSION THREAD

**Discussion by mcuiobroker** [Zefau/ioBroker.jarvis#2783] [General]
@zefau 
Kannst du bitte genau erklären, warum nun v3.2.0 auf Port 840X umgestellt werden musste?
Warum gibt / gab es Probleme mit Port 8082?
Ich würde es dann gerne in die Doku mit aufnehmen.
Bitte dabei berücksichtigen, dass durch die Umstellung die Skripte teilweise umständlich geladen werden müssen, mit Upload der Skripte + Upload von jarvis.
Und das die VISU-App nicht mehr genutzt werden kann. -> Apple iOS User können jetzt jarvis nicht mehr von unterwegs nutzen? Geht bei mir auf jeden Fall nicht.
Da wäre dann die Frage wann ist mit der Jarvis - iOS-App zu rechnen?

Vielen Dank

**Comment by Zefau** [comment]
Um reserve proxy fehlerfrei möglich zu machen und die neue Authentifizierung sauberer technisch zu integrieren

**Comment by godfuture** [comment]
Ebenfalls eine Frage zum Port. Wieso muss der Port 8400 von Host zu Container freigegeben werden? Jarvis läuft doch im selben Container sprich Netzwerk.

Aktuell sieht es so aus, als ginge Jarvis auf den Host und vom Host wieder in den Container. Localhost, local docker IP oder 127.0.0.1 gehen mit 8400 nicht. 

Wieso ist das der Fall? Vielen Dank!

**Comment by mcuiobroker** [comment]
ChatGPT:

🔍 Warum muss Port 8400 vom Host zum Container freigegeben werden, obwohl Jarvis im selben Container läuft?
👉 Kurzfassung:
Weil Jarvis im Browser (Client) läuft und dein Browser nicht im Container ist, sondern auf dem Host oder einem anderen Gerät – und daher den Port von außen (Hostnetzwerk) erreichbar braucht.

🧠 Detaillierte Erklärung
💡 Jarvis ist eine Web-Frontend-App, kein reiner Server
Jarvis (iobroker.jarvis) besteht aus:

einem Backend, das mit ioBroker kommuniziert

einem Frontend (Web-App), das der Browser über HTTP/S vom Host lädt

Wenn du http://<host-ip>:8400 im Browser aufrufst, dann greift nicht Docker intern, sondern dein Hostnetzwerk/Browser auf diesen Port zu

📦 Docker Container-Netzwerk ≠ Hostnetzwerk
Wenn du den ioBroker + Jarvis in einem Container betreibst:

Der Port 8400 ist intern im Container erreichbar

Aber nicht automatisch vom Host aus, außer du veröffentlichst ihn

Beispiel docker run:

```
docker run -p 8400:8400 iobroker/iobroker:latest
```
Nur mit -p 8400:8400 wird der interne Port 8400 im Container auf den Port 8400 deines Hostsystems weitergeleitet.

❌ Warum gehen localhost:8400 oder 127.0.0.1 im Container nicht?
Weil:

localhost im Browser bezieht sich auf den Rechner des Nutzers (also dein Host-PC, nicht den Container!)

127.0.0.1 im Container ist nur der Container selbst

Docker-Container laufen standardmäßig in einem isolierten Bridge-Netzwerk

Du brauchst daher entweder:

✅ Port-Freigabe per -p
oder
✅ Netzwerkmodus host (nicht empfohlen für jeden Anwendungsfall)
